src/app.component.tsx

```diff
@@ -15,6 +16,34 @@ export function AppComponent() {
   const [showUserLocation, setShowUserLocation] = useState(false)
   const [selectedRoutes, setSelectedRoutes] = useState<Set<string>>(new Set())
 
+  useEffect(() => {
+    let lang = localStorage.getItem('language')
+    if (!lang) {
+      let client = new XMLHttpRequest()
+      client.open('GET', window.location.origin, true)
+      client.send()
+
+      client.onreadystatechange = function() {
+        if (this.readyState === this.HEADERS_RECEIVED) {
+          let serverLang = client.getResponseHeader('Accept-Language')
+          // let serverLang = 'en-US,en;q=0.9,ar;q=0.8,de;q=0.7,ko;q=0.6,it;q=0.5,nl;q=0.4,pt;q=0.3,ru;q=0.2,es;q=0.1,ga;q=0.1,af;q=0.1,ca;q=0.1,gd;q=0.1,fr;q=0.1';
+          if (!serverLang) {
+            localStorage.setItem('language', 'en')
+            i18n.changeLanguage('en')
+          } else {
+            serverLang = serverLang.split(',')[0].split('-')[0]
+            if (serverLang in resources) {
+              localStorage.setItem('language', serverLang)
+              i18n.changeLanguage(serverLang)
+            }
+          }
+        }
+      }
+    } else {
+      i18n.changeLanguage(lang)
```

Comment:
Too much nested `if` conditions.  
There should be `i18n.changeLanguage(lang)` called only once.

```js
const language = localStorage.getItem('language') || navigator.language.split('-')[0]
const existingLanguage = resources[language] ? language : 'en'
localStorage.setItem('language', existingLanguage)
i18n.changeLanguage(existingLanguage)
```